src/Icons/src/IconPreRendererInterface.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons;
+
+interface IconPreRendererInterface
+{
+    public function __invoke(string $name, Icon $icon): Icon;
```

Comment:
This is not possible, there is no plan to expose the Icon object for now :/ 

Do you see another way ?

Reply:
Icon object is currrntly passed down to prerenders without my changes,so I don't understand you and the reasoning

Reply:
The "prerenders" are currently internal functions hard-coded in the renderer. Currently again, this is nothing else than an implementation detail, that could change at any time. 

So i'm open to discuss both, but it's not at all a minor thing

And if we expose the Icon object (internal for now) here, we both know this is a question of days before beeing asked, logically, to expose it elsewhere..

Reply:
Okay, so what do you prefer to expose if not the icon object? Why current prerenderers are getting the icon object?

Reply:
I'm sorry this is me not understanding right now... 

You are talking about prerenderers like it was a thing documented, exposed, etc.

This  "current prerender"  (no plural) is right now  _one_ callable that is part of the _internal_ mecanism of the IconRenderer, in a `private` method. 

Again, i'm not against discussing it, but it seems fair to describe the current state objectively :)

Reply:
I can live with it if you decide to pass only attributes,but I see no harm if we'd have like I've suggested.
```php
function __invoke(array $attributes) {
  $attributes['test'] = 'something';

  return $attributes;
}
```
vs
```php
function __invoke(Icon $icon) {
  return $icon->withAttributes(['test' => 'something']);
}
```

To me the second solution looks more appealing than the first one, because intention is to modify the icon :)

Reply:
> I'm sorry this is me not understanding right now... 
> 
> You are talking about prerenderers like it was a thing documented, exposed, etc.
> 
> This  "current prerender"  (no plural) is right now  _one_ callable that is part of the _internal_ mecanism of the IconRenderer, in a `private` method. 
> 
> Again, i'm not against discussing it, but it seems fair to describe the current state objectively :) 

Yeah, but you decided to pass the Icon, not me:)

Reply:
To a private internal function ;) And i know you know very well the [difference](https://symfony.com/doc/current/contributing/code/bc.html) 😉

Reply:
So indeed it would probably be (not saying "ok" for now, we need to discuss the impacts with the team) something like 

```
callable(string, array): array 
```

with the icon name and the attributes passed during runtime

These attributes would then be added to the current ones..

But i'm realizing.... why don't just decorate the IconRenderer ?

Reply:
Just pushed this example of decoration (i'm sure you don't need an example to know how to do this, but if anyone comes reading this exchange ;) )

https://github.com/symfony/ux/blob/cd7cf7ff7c8db4f3aba54add9ee046df4a9025d9/ux.symfony.com/src/UX/IconRenderer.php